*** propgrid.cpp.orig	2011-05-11 16:26:26.000000000 +0200
--- propgrid.cpp	2011-11-21 15:56:12.020787114 +0100
***************
*** 708,715 ****
  {
  #if 1  // defined(__WXPYTHON__)
      delete m_OORClientData;
!     if ( m_clientData )
          Py_DECREF( m_clientData );
  #endif
  
      DoEmpty();  // this deletes items
--- 708,718 ----
  {
  #if 1  // defined(__WXPYTHON__)
      delete m_OORClientData;
!     if ( m_clientData ) {
! 		PyGILState_STATE state = PyGILState_Ensure();
          Py_DECREF( m_clientData );
+         PyGILState_Release(state);
+ 	}
  #endif
  
      DoEmpty();  // this deletes items
***************
*** 1268,1274 ****
--- 1271,1279 ----
      {
          if ( obj == Py_None )
              value.MakeNull();
+         PyGILState_STATE state = PyGILState_Ensure();
          Py_DECREF(obj);
+         PyGILState_Release(state);
      }
  #endif
  
***************
*** 9602,9626 ****
      }
      wxPGVariantDataPyObject( PyObject* value )
      {
          if (!value) value = Py_None;
          Py_INCREF(value);
          m_value = value;
      }
      virtual ~wxPGVariantDataPyObject()
      {
          // Avoid crashing on program exit
!         if ( _PyThreadState_Current != NULL && m_value )
              Py_DECREF(m_value);
      }
      inline PyObject* GetValue() const { return m_value; }
      inline PyObject* GetValueRef() const { return m_value; }
      inline void SetValue(PyObject* value)
      {
          if (m_value)
              Py_DECREF(m_value);
          if (!value) value = Py_None;
          Py_INCREF(value);
          m_value = value;
      }
      // TODO
      virtual bool Eq(wxVariantData&) const { return false; }
--- 9607,9638 ----
      }
      wxPGVariantDataPyObject( PyObject* value )
      {
+ 		PyGILState_STATE state = PyGILState_Ensure();
          if (!value) value = Py_None;
          Py_INCREF(value);
          m_value = value;
+         PyGILState_Release(state);
      }
      virtual ~wxPGVariantDataPyObject()
      {
          // Avoid crashing on program exit
!         if ( _PyThreadState_Current != NULL && m_value ) {
! 			PyGILState_STATE state = PyGILState_Ensure();
              Py_DECREF(m_value);
+             PyGILState_Release(state);
+ 		}
      }
      inline PyObject* GetValue() const { return m_value; }
      inline PyObject* GetValueRef() const { return m_value; }
      inline void SetValue(PyObject* value)
      {
+ 		PyGILState_STATE state = PyGILState_Ensure();
          if (m_value)
              Py_DECREF(m_value);
          if (!value) value = Py_None;
          Py_INCREF(value);
          m_value = value;
+         PyGILState_Release(state);
      }
      // TODO
      virtual bool Eq(wxVariantData&) const { return false; }
***************
*** 9639,9648 ****
--- 9651,9662 ----
  
  PyObject* operator <<( PyObject* value, const wxVariant &variant )
  {
+ 	PyGILState_STATE state = PyGILState_Ensure();
  	wxPGVariantDataPyObject *data = wxDynamicCastVariantData( variant.GetData(), wxPGVariantDataPyObject );
      wxASSERT( data );
      value = data->GetValue();
      Py_INCREF(value);
+     PyGILState_Release(state);
      return value;
  }
  
***************
*** 9659,9665 ****
--- 9673,9681 ----
      if ( !data )
          return NULL;
      PyObject* retval = data->GetValue();
+     PyGILState_STATE state = PyGILState_Ensure();
      Py_INCREF(retval);
+     PyGILState_Release(state);
      return retval;
  }
  
